types/experiments.ts

```diff
@@ -0,0 +1,81 @@
+/** All fields are publically accessible via the API and available to clients */
+interface Experiment {
+  /** Unique identifier for the experiment */
+  slug: string;
+  /** Publically-accesible name of the experiment */
+  name: string;
+  /** Short public description of the experiment */
+  public_description: string;
+
+  /** Current state of the experiment. */
+  status: "Draft" | "Review" | "Accepted" | "Rejected" | "Live" | "Complete";
+  /** Is the experiment currently live in production? i.e., published to remote settings? */
+  isPublished: boolean;
+  /** Are we continuing to enroll new users into the experiment? */
+  isEnrollmentPaused: boolean;
+
+  /** Reference to an Audience, which contains targeting information */
+  audience: Audience;
+  /** Bucketing configuration */
+  bucketConfig: BucketConfig;
+  /** A list of cusom feature metrics relevant to the experiment analysis */
+  featureMetrics: Array<FeatureMetric>;
+  /** Branch configuration for the experiment */
+  branches: Array<Branch>;
+
+  /** Experimenter URL. Generated by Experimenter. */
+  experiment_url: string;
+  /** Actual publish date of the experiment */
+  start_date: Date;
+  /** Actual end date of the experiment */
+  end_date: Date;
+}
+
+/** These fields are internal to Experimenter */
+interface ExperimentDetails extends Experiment {
+  hypothesis: string;
+  namespace?: string;
+  proposed_start_date: Date;
+  proposed_end_date: Date;
+  proposed_enrollment: number;
+}
+
+interface BucketConfig {
+  randomization_unit: "client_id";
+  inputs: Array<string>
+  start: number;
+  count: number;
+  total: number;
+}
+
+type TargetingExpression = {
+  id: string;
+  description: string;
+  value: string;
+}
+
+type AudienceConditionGroup = TargetingExpression | ["and" | "or", ...Array<AudienceConditionGroup>];
+
+interface Audience {
+  id: string;
+  name: string;
+  description: string;
+  conditions: AudienceConditionGroup;
+}
+
+interface Branch {
+  slug: string;
+  ratio: number;
+  // This is weird, let's make this less weird
+  groups: Array<string>;
+  value: null | MessageConfig | boolean | string | number;
+}
+
+// TODO - Needs to import all messaging types
+interface MessageConfig {
+  [key: string]: any;
```

Comment:
Note: importing between files in this directory currently doesn't work. That's something I hope to improve in the future.